source/common/grpc/async_client_impl.cc

```diff
@@ -254,7 +254,7 @@ void AsyncRequestImpl::cancel() {
 }
 
 void AsyncRequestImpl::onCreateInitialMetadata(Http::RequestHeaderMap& metadata) {
-  current_span_->injectContext(metadata);
+  current_span_->injectContext(metadata, *options_.parent_context.stream_info);
```

Comment:
This is quite complicated, and we do not necessarily want the destination IP address in an async request. `addressUsedAtClient` in SkyWalking is not a must to be an IP address. So I think we can fall back to the traditional behavior here. WDYT?

Reply:
But now that we're going to modify this API, we should try to keep it correct in all scenarios as much as possible. But I admit that we don't need to use this one PR to solve all the problems.

Reply:
Here it injects authority to prop header, the behavior won't be replaced with this change only for async client. I think it will be processed as future work.

Reply:
Get it. I will recommend to update the API to something like this:

```
  virtual void injectContext(TraceContext& trace_conext,
                             const Upstream::HostDescriptionConstSharedPtr& upstream) PURE;
```

For the async client, just keep upstream to be nullptr.

Reply:
much cleaner, thanks.